### PR TITLE
[FFI] Handling the default library loading on z/OS

### PR DESCRIPTION
**Pull request by ChengJin01**
The changes follow the same way of accessing the default
libraries on AIX with the inlined functions in a list to
address the issue with the default libraries during the
FFI downcall on z/OS.

Related: [Internal 441](https://github.ibm.com/runtimes/javanext/issues/441)

Signed-off-by: ChengJin01 <jincheng@ca.ibm.com>

### CODE REVIEW

**Comment by ChengJin01** [comment]
The PR (verified on AIX) has the same changes as https://github.com/ibmruntimes/openj9-openjdk-jdk21/pull/159.

**Review by keithc-ca — CHANGES_REQUESTED** [review]

File: src/java.base/share/classes/jdk/internal/foreign/SystemLookup.java [hunk 1]
```diff
@@ -82,16 +87,38 @@ private static SymbolLookup makeSystemLookup() {
...
 
+    /* Obtain the address of the inlined function list by loading the specified library. */
+    private static MemorySegment getInlinedFunctListAddr() {
+        /* Directly load the specified library only once with the inlined libc functions
+         * missing in the default library.
+         */
+        SymbolLookup funcsLibLookup = libLookup(libs -> libs.load(jdkLibraryPath("syslookup")));
+
+        /* Perform a FFI downcall to obtain the address of the inlined function list. */
+        MemorySegment funcs_addr = funcsLibLookup.find("funcs_addr").orElseThrow();
```
[the remark this answers is not in the document]
Comment (ChengJin01) [reply src/java.base/share/classes/jdk/internal/foreign/SystemLookup.java]: Fixed as suggested above.

**Comment by keithc-ca** [comment]
Jenkins test sanity aix,alinux jdknext

**Comment by ChengJin01** [comment]
The failure at https://openj9-jenkins.osuosl.org/job/Test_openjdknext_j9_sanity.functional_aarch64_linux_Personal/37/artifact/Test_openjdknext_j9_sanity.functional_aarch64_linux_Personal_testList_0/37/aqa-tests/TKG/output_17168469728045/Test_openjdknext_j9_sanity.functional_aarch64_linux_Personal_testList_0.tap
```
        Caused by: java.util.NoSuchElementException: Symbol not found: funcs_addr
        	at java.base/java.lang.foreign.SymbolLookup.findOrThrow(SymbolLookup.java:179)
        	at java.base/jdk.internal.foreign.SystemLookup.getInlinedFunctListAddr(SystemLookup.java:98)
        	at java.base/jdk.internal.foreign.SystemLookup.<clinit>(SystemLookup.java:65)
        	... 34 more
        -----------------------------------
        Jep454Tests_testLinkerFfi_DownCall_0_FAILED
```
indicates we should do nothing in `getInlinedFunctListAddr()` on other platforms give `funcs_addr` is specific to AIX & z/OS. So I suggest to add a piece of code to simply return null for non-AIX/zOS platform in the static method as follows:
```
    private static MemorySegment getInlinedFunctListAddr() {
   if (!OperatingSystem.isAix() || !OperatingSystem.isZOS()) {
         return null;
     }
    ....
}
```

**Comment by keithc-ca** [comment]
> simply return null for non-AIX/zOS platforms

I suggest that should happen within `getInlinedFunctListAddr()` as it already checks for AIX; so moving the definition of `funcsLayout` to the beginning addresses the problem:
```java
        SequenceLayout funcsLayout;

        if (OperatingSystem.isAix()) {
            funcsLayout = AixFuncSymbols.LAYOUT;
        } else if (OperatingSystem.isZOS()) {
            funcsLayout = ZosFuncSymbols.LAYOUT;
        } else {
            return null;
        }
```

**Comment by keithc-ca** [comment]
Jenkins test sanity aix,alinux jdknext

**Review by keithc-ca — APPROVED** [review]

**Pull request merged**